packages/sdk-browser/src/utils.ts

```diff
@@ -28,11 +28,30 @@ export const createEmbeddedAPIKey = async (
     aead: AeadId.Aes256Gcm,
   });
 
+  // Function to extract x and y components from a hex string public key
+  function getXYComponentsFromHexString(publicKeyHex: string) {
+    publicKeyHex = publicKeyHex.replace(/^0x/, '');
+    if (!publicKeyHex.startsWith('04')) {
+      throw new Error("Public key is not in uncompressed format");
+    }
+    const x = publicKeyHex.slice(2, 66);
+    const y = publicKeyHex.slice(66, 130);
+    return { x, y };
+  }
+
   // 3: import the targetPublicKey (i.e. passed in from the iframe)
   const targetKeyBytes = uint8ArrayFromHexString(targetPublicKey);
+  const { x, y } = getXYComponentsFromHexString(targetPublicKey);
+
   const targetKey = await crypto.subtle.importKey(
-    "raw",
-    targetKeyBytes,
+    "jwk",
+    {
+      kty: "EC",
+      crv: "P-256",
+      x: base64UrlEncode(uint8ArrayFromHexStringPadded(x, 32)),
```

Comment:
Down to call this `uint8ArrayFromHexStringWithPadding`, or add the length argument as optional to the existing `uint8ArrayFromHexString` implementation